knowledge graph/kg_agent.py

```
@@ -0,0 +1,326 @@
+    async def query_graph(self, query: str, params: Dict[str, Any] = None):
+        """Run a Cypher query against the knowledge graph"""
+        async with self.driver.session() as session:
+            result = await session.run(query, params or {})
+            records = await result.fetch_all()
+            return [dict(record) for record in records]
+    
+    async def get_related_entities(self, entity_type: str, properties: Dict[str, Any], depth: int = 1):
+        """Get entities related to a specific entity up to a certain depth"""
+        async with self.driver.session() as session:
+            # build match clause
+            props_str = " AND ".join([f"e.{k} = ${k}" for k in properties.keys()])
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)